### ISSUE TITLE
Wrong colors in Tandy 4 color palette

### ISSUE THREAD

**Issue by Fenix770** [86Box/86Box#510]
Explanation 

The game "Amazing of Spiderman" has a separated graphics mode for Tandy 4 color and 16 colors and other for CGA 4 colors. In standard IBM PC with CGA/EGA/VGA card or CGA emulation used in the Hercules card, the Tandy 4 color video mode looks identical to CGA. But In Tandy 1000 machine  the Tandy 4 colors option has a different palete of colors than CGA.

**Bug:** Dosbox seems to detect and use the alternate 4 colors palette if the machine type is set to Tandy and the option 3 is selected. In 86box in both Tandy 1000 HX and SL/2 emulation the standard CGA pallete is used insted of the Tandy palette even with if the option 3 is selected.

Here is the screenshot of the video modes. 
![spidey_002](https://user-images.githubusercontent.com/31600563/71533550-1fd01900-28d8-11ea-981e-9a15971e5a9a.png)

Here is the game with the Tandy 4 colors mode with Tandy emulation in Dosbox
![spidey_000](https://user-images.githubusercontent.com/31600563/71533584-5148e480-28d8-11ea-954d-0e7f66df5345.png).

And here the CGA 4 colors mode or Tandy 4 colors mode in non Tandy 1000 machine or 86box
![spidey_001](https://user-images.githubusercontent.com/31600563/71533640-b997c600-28d8-11ea-8d4c-a190dc1b599a.png)

Here is image file of the game: 
[Amazing Spiderman.zip](https://github.com/86Box/86Box/files/4006317/Amazing.Spiderman.zip)

The game starts with Spidey.exe file.

_Posible reason for 4 color mode on Tandy 1000:_ The Tandy 4 color video mode could be necessary only in Tandy 1000 machines with 384kb of ram, because 128kb was used by the video chip when a 16 color video mode is used leaving insuficient ram for the game itself. Althogh I did not test it in a real Tandy machine, some machines sold in 1988 had 384kb ram like EX,HX and SL. Later models had 512kb as minimum system memory.

**Comment by OBattler** [comment]
I know that the CGA, if I recall correctly, has a third color palette with exactly the red, cyan, and white colors. What happens if you try to select the Tandy 4 colors mode in the game on a CGA on 86Box?

**Comment by OBattler** [comment]
OK, I fixed this now and committed the fix. It turns out the Tandy Graphics Adapter processes the CGA colors through the palette array after doing the normal CGA stuff, and that wasn't implemented.

**Comment by OBattler** [comment]
Since this is fixed, I'm closing this.

**Comment by dexpd4** [comment]
I would be intrested in re-creating this bug.  Is there a copy anywhere of a pre v2.10 86Box?  I had an old IBM with Dosshell and used to play the old gold box games(pools of radiance, curse of the azure bonds, secret of the silver blades) it always asked me to select a graphics card before starting the game and if I selected Tandy I got similar results on the colors.  I always thought there was a computer out there that was designed to make everything pink and cyan XD
